python/tk_framework_desktopserver/shotgun/scripts/cache_commands.py

```diff
@@ -12,9 +12,27 @@
 import cPickle
 import glob
 import os
+import sqlite3
+import contextlib
 
 def cache(cache_file, data, base_configuration, engine_name, config_data):
-    import sqlite3
+    """
+    Populates the sqlite cache with a row representing the desired pipeline
+    configuration and entity type. If an entry already exists, it is updated.
+
+    :param str cache_file: The path to the sqlite cache file on disk.
+    :param dict data: The raw payload send down by the client.
+    :param str base_configuration: The desired base pipeline configuration's
+        uri.
+    :param str engine_name: The name of the engine to bootstrap into. This
+        is most likely going to be "tk-shotgun"
+    :param dict config_data: All relevant pipeline configuration data. This
+        dict is keyed by pipeline config entity id, each containing a dict
+        that contains, at a minimum, "entity", "lookup_hash", and
+        "contents_hash" keys.
+    """
+    # The local import of sgtk ensures that it occurs after sys.path is set
+    # to what the server sent over.
     import sgtk
```

Comment:
This bootstrapping is bug prone. You are bringing the core that bootstrap into the scope and the handle is still available after bootstrapping. For this method there's no real harm since you don't use sgtk after bootstrapping, but moving the instantiation of the manager to a separate method could avoid bugs in the future if we ever need to call methods on sgtk module after the bootstrap.

Also,  I just noticed that you are going multiple bootstraps in the loop. I am worried this might have some weird side-effects, but I don't have a solution to offer to be honest. We are using local imports in quite a few places in Toolkit, which means that the call from the manager might be doing some local imports which will then jump into the updated version of the library instead of the original one.

I think @manneohrstrom this qualifies for the worries we had a while back with the core swapping. :)